javalib/src/main/scala/java/io/File.scala

```diff
@@ -264,7 +264,7 @@ class File(_path: String) extends Serializable with Comparable[File] {
 
   def list(filter: FilenameFilter): Array[String] =
     if (!isDirectory() || !canRead()) {
-      null
+      Array.empty[String]
```

Comment:
If JVM returns null in this case we should return null too, we can not change the semantics.